enhancements/sig-policy/99-policy-placement-strategy/README.md

```diff
@@ -0,0 +1,302 @@
+
+Additionally, policies have a `remediationAction` specified, which can either be `inform`, meaning the policy controller
+monitors objects on the managed cluster reporting status without making changes, or `enforce`, meaning the policy
+controller attempts to make changes on the managed cluster based on the policy definition and reports status accordingly
+based on the success or failure of the update.
+
+### Design
+
+For this rollout enhancement, policy propagation by the `governance-policy-propagator` controller on the hub would take
+place in a multiple-pass approach:
+
+1. On the first pass following a root policy creation or update, the `governance-policy-propagator` controller on the
+   hub cluster will replicate all policies to managed cluster namespaces and are created regardless of the rollout
+   strategy specified, with the remediation action set to `inform`. This way, all policies are up-to-date and reporting
+   a compliance status based on the current version of the root policy. The aggregated rollout status on the root policy
+   is set to `Progressing`. If the remediation action on the root policy is already `inform`, the rollout status on each
+   cluster is set to `Progressing` and no second pass occurs. If the remediation action is `enforce`, the rollout status
+   on each cluster is set to `ToApply`.
+2. On the subsequent passes, if the `remediationAction` is `enforce`, the `governance-policy-propagator` fetches the
+   managed clusters returned from the `Placement` library (based on the user-configured rollout strategy) and sets the
+   rollout status of each returned cluster to `Progressing`, updating any `remediationAction` to `enforce` as defined in
+   the root policy.
+
+From here, work is picked up by the `governance-policy-framework` controllers on the managed clusters:
+
+1. The template sync controller of the framework updates the applicable `ConfigurationPolicy` objects like usual, which
+   triggers reevaluation from the `config-policy-controller`.
+2. Once the `ConfigurationPolicies` have a generation that matches the `lastEvaluatedGeneration` set on the policy
+   status, the status sync controller of the framework will set the rollout status to to `Succeeded` after all of the
+   `ConfigurationPolicies` have a compliant status or `Failed` if there is any non-compliant status.
+
+### User Stories
+
+#### Story 1
+
+As a system administrator, I want to know the status of a rollout for a policy and what clusters have been updated.
+
+- **Summary**
+
+  - Add a `RolloutStatus` to the `Policy` status in the CRD.
+
+- **Details**
+
+  - The `RolloutStatus` would be added to reflect: rollout status on the replicated policy, per-cluster on the root
+    policy, and an aggregated status on the root policy. The aggregated status would specify: `Progressing` (this would
+    include `ToApply`), `Succeeded`, `Failed`, or (this would include `TimeOut`).
```

Comment:
I think `ToApply` means the the policy hasn't been enforced yet on the cluster.

`Progressing` would mean the policy has been enforced on the cluster but the rollout hasn't completed yet.

Reply:
Is the set of "valid" status fields different on the root policy vs the replicated policies? It sounds like ToApply would be valid on a replicated policy but not the root. Similarly it sounds like progressing would be valid on the root but not the replicated set. If so it might help to define the set of valid values for root vs replicated.

Reply:
Good point, @mprahl--I wrapped `ToApply` in for simplicity but it would be clearer to use `ToApply`.

@imiller0 The `RolloutStatus` is an iota we're taking from the Placement library: https://github.com/open-cluster-management-io/api/blob/9675ab737d2086dd7094fc962c9617cc71ec7484/cluster/v1alpha1/helpers.go#L21-L38 but having these well defined for each the root and replicated policies would be good.